src/http/server.cr

```diff
@@ -138,8 +138,8 @@ class HTTP::Server
     end
   end
 
-  def bind
-    @server ||= TCPServer.new(@host, @port)
+  def bind(reuse_port = false)
```

Comment:
Wouldn't it make more sense to have either all of host, port and reuse_port on `#bind`, or all in either `.new` or `#listen`. I appreciate it makes the API a bit more unwieldy but it seems to me that `reuse_port` option should be in the same place as `port` and `host`.